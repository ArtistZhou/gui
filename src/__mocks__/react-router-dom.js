import React from 'react';
import { compose, setDisplayName } from 'recompose';
import * as rrd from 'react-router-dom';
// Just render plain div with its children
rrd.BrowserRouter = compose(setDisplayName('BrowserRouter'))(({ children }) => <div>{children}</div>);

module.exports = rrd;
