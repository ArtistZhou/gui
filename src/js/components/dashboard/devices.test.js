import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { createMount } from '@material-ui/core/test-utils';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Devices from './devices';

const mockStore = configureStore([]);
const store = mockStore({
  deployments: {
    deploymentDeviceLimit: 5000
  },
  devices: {
    byId: {},
    byStatus: {
      accepted: {
        total: 0,
        deviceIds: []
      },
      inactive: {
        total: 0
      },
      pending: {
        total: 0
      }
    }
  },
  users: {
    onboarding: { complete: false },
    showHelptips: true
  }
});

it('renders correctly', () => {
  const tree = createMount()(
    <MemoryRouter>
      <Provider store={store}>
        <Devices />
      </Provider>
    </MemoryRouter>
  );
  expect(tree.html()).toMatchSnapshot();
});
