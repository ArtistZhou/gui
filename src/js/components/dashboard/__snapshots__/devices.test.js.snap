// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h4
    className="dashboard-header"
  >
    <span>
      Devices
    </span>
  </h4>
  <div
    style={
      Object {
        "marginBottom": "30px",
        "marginTop": "50px",
      }
    }
  >
    <div
      className="notActive widget "
      onClick={[Function]}
    >
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <div
          className="widgetHeader"
          style={
            Object {
              "borderBottomStyle": "solid",
              "display": "flex",
              "flexDirection": "row",
            }
          }
        >
          <div
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
              }
            }
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-1 warning"
              focusable="false"
              role="presentation"
              style={
                Object {
                  "height": "14px",
                  "marginRight": "10px",
                  "width": "14px",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
              />
            </svg>
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                }
              }
            >
              <div
                className="hint"
              >
                 devices may be offline
              </div>
              <div
                className="tiny"
              >
                Inactive for past 24h
              </div>
            </div>
          </div>
        </div>
        <div
          className="widgetMainContent align-right"
          style={
            Object {
              "alignItems": "flex-end",
              "display": "flex",
              "flexDirection": "column",
            }
          }
        >
          <div
            className="header"
          >
            Accepted devices
          </div>
          <div
            className="counter"
          />
        </div>
        <span
          className="link"
        />
      </div>
    </div>
    <div
      className="onboard widget"
      onClick={[Function]}
    >
      <div>
        <p
          className="muted"
        >
          Learn how to connect a device
        </p>
      </div>
      <a
        className="MuiButtonBase-root-36 MuiButton-root-10 MuiButton-text-12 MuiButton-flat-15"
        href="/devices"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButton-label-11"
        >
          Connect a device
        </span>
        <span
          className="MuiTouchRipple-root-39"
        />
      </a>
    </div>
  </div>
</div>
`;
